armi/reactor/blocks.py

```diff
@@ -1518,6 +1534,87 @@ def coords(self, rotationDegreesCCW=0.0):
             round(y, units.FLOAT_DIMENSION_DECIMALS),
         )
 
+    def _createHomogenizedCopy(self, pinSpatialLocators=False):
+        """
+        Create a new homogenized copy of a block that is less expensive than a full deepcopy.
+
+        Notes
+        -----
+        This can be used to improve performance when a new copy of a reactor needs to be
+        built, but the full detail of the block (including component geometry, material,
+        number density, etc.) is not required for the targeted physics solver being applied
+        to the new reactor model.
+
+        The main use case is for the uniform mesh converter (UMC). Frequently, a deterministic
+        neutronics solver will require a uniform mesh reactor, which is produced by the UMC.
+        Many deterministic solvers for fast spectrum reactors will also treat the individual
+        blocks as homogenized mixtures. Since the neutronics solver does not need to know about
+        the geometric and material details of the individual child components within a block,
+        we can save significant effort while building the uniform mesh reactor with the UMC
+        by omitting this detailed data and only providing the necessary level of detail for
+        the uniform mesh reactor: number densities on each block.
+
+        .. note: Individual components within a block can have different temperatures, and this
+        can affect cross sections. This temperature variation is captured by the lattice physics
+        module. As long as temperature distribution is correctly captured during cross section
+        generation, it doesn't need to be transferred to the neutronics solver directly through
+        this copy operation.
+
+        .. note: If you make a new block, you must add it to an assembly and a reactor.
+
+        See Also
+        --------
+        armi.reactor.converters.uniformMesh.UniformMeshGeometryConverter.makeAssemWithUniformMesh
+        """
+
+        b = self.__class__(self.getName(), height=self.getHeight())
+        b.setType(self.getType(), self.p.flags)
+
+        # assign macros and LFP
+        b.macros = self.macros
+        b._lumpedFissionProducts = self._lumpedFissionProducts
+        b.p.buGroup = self.p.buGroup
+
+        hexComponent = Hexagon(
+            "homogenizedHex",
+            "_Mixture",
```

Comment:
@mgjarrett Can you help me understand why this new `_Mixture` material is needed here? I see that it is just a subclass of `Material` with no differences. Couldn't we have just built a Hexagon with `Material` instead?